transformations/back_translation_ner/transformation.py

```diff
@@ -0,0 +1,130 @@
+import itertools
+from typing import List
+
+import numpy as np
+from transformers import FSMTForConditionalGeneration, FSMTTokenizer
+
+from interfaces.TaggingOperation import TaggingOperation
+from tasks.TaskTypes import TaskType
+
+
+class BackTranslationNER(TaggingOperation):
```

Comment:
In my opinion, this is very similar to the regular backtranslation module; there is an improvement but the use-case is fairly limited. Would it be possible to have an NER-module that automatically tags the tokens?

Reply:
> In my opinion, this is very similar to the regular backtranslation module; there is an improvement but the use-case is fairly limited. Would it be possible to have an NER-module that automatically tags the tokens?

I would argue that (domain-specific) NER is a non-trivial problem and therefore, generic NER-modules such as spaCy entity tagger etc., cannot be used as they only work for generic entities. The entity types will even vary for different domains (just consider the domains like biomedical, material sciences etc.).

Also, the [interface](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/interfaces/TaggingOperation.py#L21) for `TaggingOperation` expects the token and tag list as input (perhaps for the very same reason that (domain-specific) NER can get tricky).

Reply:
Okay, so then if I understand your implementation correctly, it is to protect the integrity of the entities. That is a thing actually. Thanks for clarifying!